kvbc/src/categorization/versioned_kv_category.cpp

```diff
@@ -170,6 +170,21 @@ std::unordered_map<BlockId, std::vector<std::string>> VersionedKeyValueCategory:
   return found;
 }
 
+std::vector<std::string> VersionedKeyValueCategory::getBlockStaleKeys(BlockId block_id, const VersionedOutput &out) {
+  std::vector<std::string> stale_keys_;
+  for (const auto &[key, flags] : out.keys) {
+    const auto latest = getLatestVersion(key);
+    ConcordAssert(latest.has_value());
+    ConcordAssertEQ(latest->deleted, flags.deleted);
```

Comment:
Based on the merkle category code, the assert we use there also makes sense here:
```
ConcordAssertLE(block_id, tagged_version->version); -> ConcordAssertLE(block_id, latest->version);
```

Reply:
fixed